### PR TITLE
Enable query pagination

### PR DESCRIPTION
**Pull request by XuanYang-cn**
Signed-off-by: yangxuan <xuan.yang@zilliz.com>

#19043

### CODE REVIEW

**Comment by XuanYang-cn** [comment]
/run-cpu-e2e
rerun ut

**Comment by XuanYang-cn** [comment]
rerun ut

File: internal/proxy/task_query.go [hunk 1]
```diff
@@ -409,46 +409,66 @@ func IDs2Expr(fieldName string, ids *schemapb.IDs) string {
...
-			ret = &milvuspb.QueryResults{
-				FieldsData: make([]*schemapb.FieldData, len(rr.FieldsData)),
+	ret.FieldsData = make([]*schemapb.FieldData, len(validRetrieveResults[0].GetFieldsData()))
+	idSet := make(map[interface{}]struct{})
+	cursors := make([]int64, len(validRetrieveResults))
+
+	if queryParams != nil && queryParams.limit != typeutil.Unlimited {
+		loopEnd = int(queryParams.limit)
+
+		if queryParams.offset > 0 {
```
Comment (xiaofan-luan) [inline internal/proxy/task_query.go:442]: if user don't set limit, then offset won't be handled?
doesn't make sense to me

**Pull request merged**